logstash-core/lib/logstash/agent.rb

```diff
@@ -131,8 +133,42 @@ def shutdown
     shutdown_pipelines
   end
 
-  def node_uuid
-    @node_uuid ||= SecureRandom.uuid
+  def id
+    return @id if @id
+
+    uuid = nil
+    if File.exists?(id_path)
+      begin
+        uuid = File.open(id_path) {|f| f.each_line.first.chomp }
```

Comment:
@ph yes! I wanted to be defensive here. Sometimes people open files with vim and mess around with them.  We def need chomp if someone opens in vim then saves since that'll add a newline.